src/tim/layout_infer/ops/op_layout_inference.cc

```diff
@@ -169,14 +174,65 @@ OpLayoutInfer::AlignPermuteVectorForMutilInputs() {
 
 void OpLayoutInfer::ReverseInputsPermuteVector() {
   for (const auto& i_src : op_->impl()->InputsTensor()) {
+    std::shared_ptr<vx::Tensor> perm_out;
     auto input_pv = context_->GetPermuteVector(i_src);
     if (!input_pv->IsAligned()) {
-      auto perm_out = InsertPermute(context_->GetMapedTensor(i_src),
-                                    input_pv->Reverse());
+      if (i_src->IsConstTensor()) {
+        perm_out = PermuteConstTensor(i_src, input_pv);
+      } else {
+        perm_out =
+            InsertPermute(context_->GetMapedTensor(i_src), input_pv->Reverse());
+      }
       context_->UpdateTensorMap(i_src, perm_out);
       context_->SetPermuteVector(i_src, MakeShared(input_pv->Rank()));
     }
   }
 }
+
+bool OpLayoutInfer::TransposeConstTensorData(
+    const std::shared_ptr<vx::Tensor>& input,
+    const std::shared_ptr<IPermuteVector>& pv, std::vector<uint8_t>& out_data) {
+  auto vx_type = vx::TranslateDataType(input->GetDataType());
+  auto type_size = vsi_nn_GetTypeBytes(vx_type);
+  uint32_t out_size = 1;
+  for (const auto& s : input->GetShape()) out_size *= s;
+  out_size *= type_size;
+  out_data.resize(out_size);
+  if (!input->GetDataRef()) {
+    return false;
+  }
+
+  std::vector<uint32_t> perm = kNHWC2NCHW;
```

Comment:
can we make this more clear? for kernel transpose, it should something like kIcHWOc2HWIcOc